src/routes/api/decks/shared/+server.ts

```diff
@@ -1,31 +1,39 @@
-import { authenticate, getUserFromEmail, pushPath, readPath, writePath } from '$lib/server/firebaseUtils';
+import { authenticate, getUserFromEmail, pushPath, readPath, getUserData } from '$lib/server/firebaseUtils';
 import { error, json } from '@sveltejs/kit';
 import type { RequestHandler } from './$types';
 import { getStripeCustomerWithSubscriptions, isSubscribedToBlendPro } from '$lib/server/subscriptionUtils';
+import { sendDeckShareEmail } from '$lib/server/emailUtils';
 
 export const POST = (async (event) => {
   const { uid: sourceUid } = await authenticate(event);
   const { deckId, targetUserEmail } = await event.request.json();
-  if (!targetUserEmail || !deckId) 
+  if (!targetUserEmail || !deckId)
     throw error(400, "No target email and/or deckId provided");
-  const { uid: targetUid } = await getUserFromEmail(targetUserEmail);
-  const [sourceCustomer, targetCustomer, deck] = await Promise.all([
+  const targetUserPromise = getUserFromEmail(targetUserEmail).then((userData) => Promise.all([userData, getStripeCustomerWithSubscriptions(userData.uid)]));
+  const [
+    { displayName: sourceUserName = 'A Blend user' },
+    [{ displayName: targetUserName }, targetCustomer],
+    sourceCustomer,
+    deck
+  ] = await Promise.all([
+    getUserData(sourceUid),
+    targetUserPromise,
     getStripeCustomerWithSubscriptions(sourceUid),
-    getStripeCustomerWithSubscriptions(targetUid),
     readPath(`/decks/user/${sourceUid}/${deckId}`)
   ]);
```

Comment:
This is admittedly hard to read but it maximizes concurrency, which is important because it's alot of API calls and the Stripe ones are slow.